server/config/config_test.go

```diff
@@ -446,68 +450,87 @@ wait-store-timeout = "120s"
 `
 	cfg := NewConfig()
 	meta, err := toml.Decode(cfgData, &cfg)
-	c.Assert(err, IsNil)
+	re.NoError(err)
 	err = cfg.Adjust(&meta, false)
-	c.Assert(err, IsNil)
+	re.NoError(err)
 
-	c.Assert(cfg.ReplicationMode.ReplicationMode, Equals, "dr-auto-sync")
-	c.Assert(cfg.ReplicationMode.DRAutoSync.LabelKey, Equals, "zone")
-	c.Assert(cfg.ReplicationMode.DRAutoSync.Primary, Equals, "zone1")
-	c.Assert(cfg.ReplicationMode.DRAutoSync.DR, Equals, "zone2")
-	c.Assert(cfg.ReplicationMode.DRAutoSync.PrimaryReplicas, Equals, 2)
-	c.Assert(cfg.ReplicationMode.DRAutoSync.DRReplicas, Equals, 1)
-	c.Assert(cfg.ReplicationMode.DRAutoSync.WaitStoreTimeout.Duration, Equals, 2*time.Minute)
-	c.Assert(cfg.ReplicationMode.DRAutoSync.WaitSyncTimeout.Duration, Equals, time.Minute)
+	re.Equal("dr-auto-sync", cfg.ReplicationMode.ReplicationMode)
+	re.Equal("zone", cfg.ReplicationMode.DRAutoSync.LabelKey)
+	re.Equal("zone1", cfg.ReplicationMode.DRAutoSync.Primary)
+	re.Equal("zone2", cfg.ReplicationMode.DRAutoSync.DR)
+	re.Equal(2, cfg.ReplicationMode.DRAutoSync.PrimaryReplicas)
+	re.Equal(1, cfg.ReplicationMode.DRAutoSync.DRReplicas)
+	re.Equal(2*time.Minute, cfg.ReplicationMode.DRAutoSync.WaitStoreTimeout.Duration)
+	re.Equal(time.Minute, cfg.ReplicationMode.DRAutoSync.WaitSyncTimeout.Duration)
 
 	cfg = NewConfig()
 	meta, err = toml.Decode("", &cfg)
-	c.Assert(err, IsNil)
+	re.NoError(err)
 	err = cfg.Adjust(&meta, false)
-	c.Assert(err, IsNil)
-	c.Assert(cfg.ReplicationMode.ReplicationMode, Equals, "majority")
+	re.NoError(err)
+	re.Equal("majority", cfg.ReplicationMode.ReplicationMode)
 }
 
-func (s *testConfigSuite) TestHotHistoryRegionConfig(c *C) {
+func TestHotHistoryRegionConfig(t *testing.T) {
+	re := require.New(t)
+	registerDefaultSchedulers()
 	cfgData := `
 [schedule]
 hot-regions-reserved-days= 30
 hot-regions-write-interval= "30m"
 `
 	cfg := NewConfig()
 	meta, err := toml.Decode(cfgData, &cfg)
-	c.Assert(err, IsNil)
+	re.NoError(err)
 	err = cfg.Adjust(&meta, false)
-	c.Assert(err, IsNil)
-	c.Assert(cfg.Schedule.HotRegionsWriteInterval.Duration, Equals, 30*time.Minute)
-	c.Assert(cfg.Schedule.HotRegionsReservedDays, Equals, uint64(30))
+	re.NoError(err)
+	re.Equal(30*time.Minute, cfg.Schedule.HotRegionsWriteInterval.Duration)
+	re.Equal(uint64(30), cfg.Schedule.HotRegionsReservedDays)
 	// Verify default value
 	cfg = NewConfig()
 	err = cfg.Adjust(nil, false)
-	c.Assert(err, IsNil)
-	c.Assert(cfg.Schedule.HotRegionsWriteInterval.Duration, Equals, 10*time.Minute)
-	c.Assert(cfg.Schedule.HotRegionsReservedDays, Equals, uint64(7))
+	re.NoError(err)
+	re.Equal(10*time.Minute, cfg.Schedule.HotRegionsWriteInterval.Duration)
+	re.Equal(uint64(7), cfg.Schedule.HotRegionsReservedDays)
 }
 
-func (s *testConfigSuite) TestConfigClone(c *C) {
+func TestConfigClone(t *testing.T) {
+	re := require.New(t)
+	registerDefaultSchedulers()
 	cfg := &Config{}
 	cfg.Adjust(nil, false)
-	c.Assert(cfg.Clone(), DeepEquals, cfg)
+	re.True(reflect.DeepEqual(cfg.Clone(), cfg))
```

Comment:
`require.Equal` is `reflect.DeepEqual` actually. See https://github.com/tikv/pd/issues/5104.